### PR TITLE
BFB-504: Added AdditiveHUDScene and HUDManager

### PR DESCRIPTION
**Pull request by Gursaanj**
**JIRA Ticket**
https://perigongames.atlassian.net/browse/BFB-504

**Description**
Created an AdditiveHUDScene to place all HUD UI and event systems. And created a corresponding HUD manager to handle and pass logic to the UI. Moved responsibility for PlayerHealthViewBehaviour (a health bar) from UserInterfaceManager To the HUDManager.

Additionally, I Removed the AmmoCounter HUD from the AdditiveWeaponScene (as it won't be used)

**Screenshots / GIFs**
image

### CODE REVIEW

File: Assets/_BForBoss/_Core/Scripts/Managers/WeaponSceneManager.cs [hunk 1]
```diff
@@ -12,15 +12,13 @@ public class WeaponSceneManager : MonoBehaviour
         private WeaponAnimationController _weaponAnimationController = null;
 
         [SerializeField] private EquipmentBehaviour _equipmentBehaviour = null;
-        [SerializeField] private AmmunitionCountViewBehaviour _ammunitionCountView = null;
```
Comment (m4rtint) [inline Assets/_BForBoss/_Core/Scripts/Managers/WeaponSceneManager.cs:15]: can you delete the script `AmmunitionCountViewBehaviour`

File: Assets/_BForBoss/_Core/Scenes/AdditiveHUDScene.unity [hunk 2]
```diff
@@ -0,0 +1,366 @@
+%YAML 1.1
```
Comment (m4rtint) [inline Assets/_BForBoss/_Core/Scenes/AdditiveHUDScene.unity:1]: Can you remove the `EventSystem` object from this scene

File: Assets/_BForBoss/_Core/Scripts/Managers/WeaponSceneManager.cs (hunk 1 above)
Comment (Gursaanj) [reply Assets/_BForBoss/_Core/Scripts/Managers/WeaponSceneManager.cs:15]: Updated

File: Assets/_BForBoss/_Core/Scenes/AdditiveHUDScene.unity (hunk 2 above)
Comment (Gursaanj) [reply Assets/_BForBoss/_Core/Scenes/AdditiveHUDScene.unity:1]: Updated

**Review by m4rtint — APPROVED** [review]
lgtm nice

**Pull request merged**